src/index.ts

```diff
@@ -72,6 +72,19 @@ export class WebRTCPlayer extends EventEmitter {
   }
 
   private async onConnectionStateChange(e) {
+
+    if (this.peer.connectionState === 'connected') {
+      const senders = await this.peer.getSenders();
+      if (senders) {
+        senders.forEach(sender => {
+          if (sender.track.readyState === 'ended') {
+            this.log('Track ended', e);
+            return;
```

Comment:
this `return` won't actually do anything, the result with or without it is the same since it just returns undefined to the `forEach()` it doesn't abort the rest of the onConnectionStateChange code.